File: fix_frame.py
import cv2
import os

video_path = "public/hero-bg-fr/44 Clever Bridesmaid Dress Ideas - Pin-44262008833296462.mp4"
output_path = "public/hero-bg-fr/frame_270.webp"

cap = cv2.VideoCapture(video_path)
fps = cap.get(cv2.CAP_PROP_FPS)
if fps <= 0: fps = 30

frame_count = 0
extracted_count = 0

while True:
    ret, frame = cap.read()
    if not ret:
        break
        
    current_time = frame_count / fps
    target_time = extracted_count / 30.0
    
    if current_time >= target_time:
        if extracted_count == 269: # The 270th frame
            # Do NOT scale the frame, and write with maximum WebP quality (100)
            cv2.imwrite(output_path, frame, [cv2.IMWRITE_WEBP_QUALITY, 100])
            print("Successfully regenerated frame_270.webp at MAXIMUM quality and full resolution.")
            break
        extracted_count += 1
        
    frame_count += 1

cap.release()

File: temp/fix_frame.js
const ffmpegPath = require('ffmpeg-static');
const { execSync } = require('child_process');
const path = require('path');

// We use the MP4 that was downloaded earlier
const videoPath = path.join(__dirname, 'public/hero-bg-fr/44 Clever Bridesmaid Dress Ideas - Pin-44262008833296462.mp4');
const outputFrame = path.join(__dirname, 'public/hero-bg-fr/frame_270.webp');

// Extract exactly the 270th frame (n=269) with NO scaling and at maximum quality (q:v 100, lossless 1)
const cmd = `"${ffmpegPath}" -y -i "${videoPath}" -vf "select='eq(n,269)'" -vframes 1 -vcodec libwebp -lossless 1 -q:v 100 "${outputFrame}"`;

console.log("Extracting high-quality final frame...");
try {
  execSync(cmd, { stdio: 'inherit' });
  console.log("Successfully extracted maximum quality frame 270!");
} catch (e) {
  console.error("Extraction failed:", e.message);
}
